src/js/components/AppVolumesListComponent.jsx

```diff
@@ -0,0 +1,175 @@
+import classNames from "classnames";
+import React from "react/addons";
+
+import AppVolumesListItemComponent
+  from "../components/AppVolumesListItemComponent";
+
+import Util from "../helpers/Util";
+
+var AppVolumesListComponent = React.createClass({
+  displayName: "AppVolumesListComponent",
+
+  propTypes: {
+    volumes: React.PropTypes.array
+  },
+
+  getInitialState: function () {
+    return {
+      sortKey: "id",
+      sortDescending: false
+    };
+  },
+
+  getCaret: function (sortKey) {
+    if (sortKey === this.state.sortKey) {
+      return (
+        <span className="caret"></span>
+      );
+    }
+    return null;
+  },
+
+  sortBy: function (sortKey) {
+    var state = this.state;
+
+    this.setState({
+      sortKey: sortKey,
+      sortDescending: state.sortKey === sortKey && !state.sortDescending
+    });
+  },
+
+  getVolumeType: function (volume) {
+    if (volume.hostPath != null) {
+      return "DOCKER";
+    }
+    if (volume.persistent != null) {
+      return "LOCAL";
+    }
+    return null;
+  },
+
+  getVolumeRow: function (volume, index) {
+    return (
+      <AppVolumesListItemComponent
+        key={index}
+        sortKey={this.state.sortKey}
+        volume={volume}/>
+    );
+  },
+
+  getVolumes: function (volumes) {
+    var state = this.state;
+
+    return volumes
+      .map((volume) => {
+        volume.type = this.getVolumeType(volume);
+        if (volume.type === "LOCAL") {
+          volume.size = volume.persistent.size;
+          delete volume.persistent;
+        }
+
+        return volume;
+      })
+      .sort(Util.sortBy(state.sortKey, state.sortDescending))
+      .map(this.getVolumeRow);
+  },
+
+  render: function () {
+    var state = this.state;
+
+    var headerClassSet = classNames({
+      "clickable": true,
+      "dropup": !state.sortDescending
+    });
+
+    var idClassSet = classNames({
```

Comment:
This and the following lines have a lot of repetition.
Could you condense this into something reusable?